### PR TITLE
ci: Switch to Node.js 20 by default. Add initial support for Node.js 22

### PR DESCRIPTION
**Pull request by netroy**
Node.js 18.x went into maintenance October 2023, and 20.x has been the LTS since then.
We should switch over our setup also to default to 20.x, while still testing `master` CI on 18 and 22.

PS: This does not change the minimum required version of node.js. That'll stay 18 until it goes EOL on 2025-04-30. 

## Review / Merge checklist
- [x] PR title and summary are descriptive

### CODE REVIEW

File: .github/workflows/ci-master.yml [hunk 1]
```diff
@@ -9,20 +9,15 @@ jobs:
   install-and-build:
     runs-on: ubuntu-latest
 
-    timeout-minutes: 30
-
-    strategy:
-      matrix:
-        node-version: [18.x, 20.x]
```
Comment (netroy) [inline .github/workflows/ci-master.yml:16]: not sure why were were building and linting on multiple versions of node. we only need to run the tests on multiple versions.

File: .github/workflows/units-tests-reusable.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ on:
         description: 'Version of node to use.'
         required: false
         type: string
-        default: '18.x'
+        default: '20.x'
```
Comment (ivov) [inline .github/workflows/units-tests-reusable.yml:15]: Do we need the quotes?

File: packages/cli/bin/n8n [hunk 3]
```diff
@@ -21,10 +21,10 @@ if (process.argv.length === 2) {
 const nodeVersion = process.versions.node;
 const nodeVersionMajor = require('semver').major(nodeVersion);
 
-if (![18, 20].includes(nodeVersionMajor)) {
+if (![18, 20, 22].includes(nodeVersionMajor)) {
```
Comment (ivov) [inline packages/cli/bin/n8n:24]: Can we update also `CONTRIBUTING.md`

> [Node.js](https://nodejs.org/en/) version 16.9 or newer is required for development purposes.

File: .github/workflows/ci-postgres-mysql.yml [hunk 4]
```diff
@@ -106,9 +106,9 @@ jobs:
     steps:
       - uses: actions/checkout@v4.1.1
       - run: corepack enable
-      - uses: actions/setup-node@v4.0.1
+      - uses: actions/setup-node@v4.0.2
         with:
-          node-version: 18.x
+          node-version: 20.x
```
Comment (ivov) [inline .github/workflows/ci-postgres-mysql.yml:111]: Can we extract and reuse this step so we have a single spot to update?

Comment (netroy) [reply .github/workflows/ci-postgres-mysql.yml:111]: I want to do that in a separate PR. [Started a while ago, but ran into some issues](https://github.com/netroy/n8n/commit/0364b079dc5f17ee0a2fdf5ab84451911904a494).

**Review by ivov — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.44.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.44.0)

**Pull request merged**